### PR TITLE
[#84] Add support for GHC 9.4.2

### LINKED ISSUE #84: Support GHC 9.4

### PR DESCRIPTION
**Pull request by german1608**
Resolves #84

Adds support for GHC 9.4.2. Additionally, the `colourista` package declaration was updated so for cabal installs it uses the new hackage version (0.1.0.2) which supports 9.2.4, and also has a convenient base bound. For `stack`, I couldn't find a stackage for it, so I just pinned the commit from https://github.com/kowainik/colourista/commit/c7b06816ab85c27439fa7afbe6d5eed0506dd884

I also added Cabal 3.8 and GHC 9.4.2 to the build matrix for the cabal job. Let me know if we can just replace the existing cabal 3.6 with 3.8, or if it's okay to use both.

### Additional tasks

- [x] Documentation for changes provided/changed: Not needed
- [x] Tests added: Existing tests suffice.
- [x] Updated CHANGELOG.md: Should I do this?

### CODE REVIEW

**Comment by german1608** [comment]
Workflow execution sample on my fork: https://github.com/german1608/iris/actions/runs/3202513553

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -11,16 +11,17 @@ on:
...
 jobs:
   cabal:
-    name: ${{ matrix.os }} / ghc ${{ matrix.ghc }}
+    name: ${{ matrix.os }} / ghc ${{ matrix.ghc }} / cabal ${{ matrix.cabal }}
     runs-on: ${{ matrix.os }}
     strategy:
       matrix:
         os: [ubuntu-latest, macOS-latest, windows-latest]
-        cabal: ["3.6"]
+        cabal: ["3.6", "3.8"]
```
Comment (chshersh) [inline .github/workflows/ci.yml:19]: We don't need to build with multiple `cabal` versions and display cabal in the job name. It's okay to use only the latest `cabal` version 👍🏻 

There's might be a reason not to do it when it doesn't work. But I expect things to be fine here.

File: iris.cabal [hunk 2]
```diff
@@ -25,7 +25,7 @@ source-repository head
   location:            https://github.com/chshersh/iris.git
 
 common common-options
-  build-depends:       base >= 4.14 && < 4.17
+  build-depends:       base >= 4.14 && < 4.18
 
   ghc-options:         -Wall
```
Comment (chshersh) [inline iris.cabal:30]: Let's also add the `-Wredundant-strictness-flags` flag as shown on the screenshot below for GHC 9.4. This is a new GHC sanity-check not enabled by default or by `-Wall` which I still find useful 🙂 

image

File: stack.yaml [hunk 3]
```diff
@@ -1,5 +1,5 @@
 resolver: nightly-2022-08-04
 
 extra-deps:
-  - github: uhbif19/colourista
-    commit: 8148a0446bf61814f79d6a0c497007fde72b31eb
+  - github: kowainik/colourista
+    commit: c7b06816ab85c27439fa7afbe6d5eed0506dd884
```
Comment (chshersh) [inline stack.yaml:5]: I don't think we need to depend on a GitHub commit and we can use the version from Hackage instead:

```yaml
- colourista-0.1.0.2
```

File: iris.cabal [hunk 4]
```diff
@@ -25,7 +25,7 @@ source-repository head
   location:            https://github.com/chshersh/iris.git
 
 common common-options
```
Comment (chshersh) [inline iris.cabal:27]: Let's also update `tested-with`

**Review by chshersh — CHANGES_REQUESTED** [review]
Thanks for improving this! I'm glad to see that not many code or dependency changes were required 🙂 

Only a few things left to improve and it's good to go 🚢

**Review by chshersh — APPROVED** [review]
Perfect 🔝

**Pull request merged**